internal/controller/utils/nim.go

```diff
@@ -189,12 +190,26 @@ func getNimRuntimes(runtimes []NimRuntime, page, pageSize int) ([]NimRuntime, er
 
 	runtimes = append(runtimes, mapNimCatalogResponseToRuntimeList(catRes)...)
 	if catRes.Params.Page < catRes.ResultPageTotal-1 {
-		return getNimRuntimes(runtimes, page+1, pageSize)
+		return getNimRuntimes(logger, runtimes, page+1, pageSize)
 	}
 
 	return runtimes, nil
 }
 
+func logApiResponse(logger logr.Logger, resp *http.Response) {
+	logger.V(1).Info(fmt.Sprintf("got api response %s", resp.Status))
+	if resp.StatusCode != http.StatusOK {
+		if resp.ContentLength > 0 {
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				logger.V(1).Error(fmt.Errorf("failed to parse body"), "")
```

Comment:
append the error here?

Reply:
Resolved in https://github.com/opendatahub-io/odh-model-controller/pull/383/commits/bd22ff20f386d309f68bfb795866b9db9c4452cc.